cmd/zoekt-indexserver/main.go

```diff
@@ -164,6 +166,61 @@ func indexPendingRepos(indexDir, repoDir string, opts *Options, repos <-chan str
 	}
 }
 
+type indexRequest struct {
+	RepositoryUrl  string // TODO: Decide if tokens can be in the URL or if we should pass separately
+	RepositoryName string
+}
+
+func startIndexingApi(repoDir string, indexDir string, indexTimeout time.Duration) {
+	http.HandleFunc("/index", func(w http.ResponseWriter, r *http.Request) {
+		dec := json.NewDecoder(r.Body)
+		dec.DisallowUnknownFields()
+		var req indexRequest
+		err := dec.Decode(&req)
+
+		if err != nil {
+			log.Printf("Error decoding index request: %v", err)
+			http.Error(w, "JSON parser error", http.StatusBadRequest)
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), indexTimeout)
```

Comment:
I think in practice we probably don't want HTTP requests blocking for 1 hour (especially from the GitLab side) so might consider making this async but it might be nice to get some acknowledgement of well-formed request first.